lib/FrontendTool/ScanDependencies.cpp

```diff
@@ -719,6 +719,18 @@ bool swift::batchScanModuleDependencies(CompilerInvocation &invok,
       SmallVector<const char*, 4> args;
       llvm::cl::TokenizeGNUCommandLine(entry.arguments, saver, args);
       CompilerInvocation subInvok = invok;
+      // Detect if Clang target arguments are present in the form of -Xcc -target -Xcc <triple>
+      // and set that to be the invocation target.
+      for (auto it = args.begin(), end = args.end(); it != end; ++it) {
+        StringRef arg = *it;
+        if (arg == "-target") {
+          StringRef next = *(it+1);
+          if (next == "-Xcc") {
+            assert(it+2 != end);
+            subInvok.setTargetTriple(*(it+2));
+          }
+        }
+      }
```

Comment:
Should we extract all these `-xcc` arguments to add to the compiler instance via `setExtraClangArgs`?

Reply:
Oh, for some reason I assumed that this would be taken care of when we pass them into `CompilerInvocation::parseArgs(`. But if it is not, then yes, we absolutely should.

Reply:
Even better, we need to teach `parseArgs` to parse these extra clang arguments properly.

Reply:
After teaching `parseArgs` to set up `ClangImporterOpts.ExtraArgs` we would still require this change though. Is it alright to do it in a followup?

Reply:
Could you clarify why we still need this change?

Reply:
It seems to me that `parseArgs()` is handling `-Xcc` arguments properly. The real issue here is when we are creating a clang instance for dependencies scanning, we never use these extra arguments (see ClangImporter::getModuleDependencies for details).

Reply:
My understanding is that even with `ClangImporterOpts.ExtraArgs` set, the target that gets used on the clang `scan-deps` invocation comes from the `ASTContext` on which we will call `getModuleDependencies`. 
That context belongs to the instance we set up here using the invocation on which we call `parseArgs`. Without this change (extracting the `-target`), this context will inherit the parent invocation's target, regardless of whether or not we correctly parsed `ExtraArgs`.

Reply:
> It seems to me that `parseArgs()` is handling `-Xcc` arguments properly. The real issue here is when we are creating a clang instance for dependencies scanning, we never use these extra arguments (see ClangImporter::getModuleDependencies for details).

Oh, yes. I think we should then make sure we go ahead and use them in `getClangDepScanningInvocationArguments`, then.

Reply:
Yeah, that sounds to me the correct solution.

Reply:
It was a bit more plumbing than I'd anticipated, but it is now ready for a review.